### ISSUE TITLE
Ability to assign catalog price per currency

### ISSUE THREAD

**Issue by mariannk** [nopSolutions/nopCommerce#1]
And contribution: http://www.nopcommerce.com/boards/t/38848/nopcommerce-380-roadmap-and-estimated-release-date-lets-discuss.aspx?p=3#158021 

At the moment prices in currencies that are different to the base currency are displayed as a direct conversion of the base price using the downloaded exchange rates. Some stores don't really work in that way as they need to be able to place themselves correctly in a foreign currency market depending on the country. I.e. for customers in Chile the price should be less than the price for customers in the UK, where standard of living is higher. This of course will need to have proper controls on the checkout to only allow purchases for customers in certain countries with certain currencies. 

How it'll be done:
There will be a radiobutton on the product details page (admin area): "Auto convert to other currencies during order processing" and "Set price manually per currency". When the second option is selected, then a store owner will be able to enter prices for all available currencies
Should we do it for tier prices, attribute (product/checkout) price adjustments, special prices, discounts, etc?
Maybe, we should create a new table (general) for storing prices similar to how it's done for localizable properties, custom fields and url records 

But it causes significant issue to the current implementation. We store prices/totals/etc for already placed orders only in the primary store currency. This way we'll have to store these values in both used currencies (now have "ExchangeRate" property) - to display proper value to customers on order details pages. It also can cause some issues when calculating discounts/totals/etc. 

Each payment gateway also should be rewritten to support distinct currencies. It causes more issue when we have some product prices specified (passed to a payment gateway) in one currency, and other product prices are in other currency (no custom price specified for certain currency abd default one is used). 

also see http://www.nopcommerce.com/boards/t/29547/nopcommerce-340-roadmap-and-estimated-release-date-lets-discuss.aspx?p=9#124517

**Comment by mohsen2hasani** [comment]
@AndreiMaz 
Hi Andrei
Can you guide me to Implement this feature?
2 month later i implement this feature for nop 3.90, because in my company, we have some custom currency and we need this feature, but, some other feature like additional prices and discounts are processed by main price.
I added currency to "Tier prices" and when admin entered price row, he can select currency ....
but i use static price code for my purpose, and if you want and help me, i want to implement this feature and commit codes for nopcommerce.

sorry for my bad English and thank you.

**Comment by shurub3l** [comment]
When is this going to be implemented? We really need this feature in the near future. I am also available to help develop it if needed.

**Comment by pixmame** [comment]
its needed to be there

**Comment by tigran-avdalyan** [comment]
@AndreiMaz
Hi Andrei
This issue is oldest on GitHub and most important, can we add to millstone 4.20?

**Comment by johansakbas** [comment]
Any input about when and if this issue will be implemented? @AndreiMaz

**Comment by adrianiftode** [comment]
Hi guys, is this something that is in progress?

**Comment by toddca** [comment]
Hey guys, This is a huge issue for a lot of us and as noted by others this is a really old issue. Please take a serious look at this issue and address for 4.2. I cannot see how you can just continue to ignore after so many years - if I am not mistaken this issue was opened prior to GitHub in 2011.

**Comment by johansakbas** [comment]
> Hey guys, This is a huge issue for a lot of us and as noted by others this is a really old issue. Please take a serious look at this issue and address for 4.2. I cannot see how you can just continue to ignore after so many years - if I am not mistaken this issue was opened prior to GitHub in 2011.

I agree with toddca. I think this is a big reason why nopCommerce isn't a more popular system overall and especially in the B2B world. nopCommerce has potential to be a first-choice, go-to platform for larger corporations that work in .NET but basic functions like this missing makes it harder IMHO.

**Comment by jgarniss** [comment]
I absolutely agree, this is necessary and has been promised for quite some time. Is this going to happen?

**Comment by toddca** [comment]
Hey guys - if you are not going to fix/address this issue I would rather someone just state as much. I don't see any attempt to try to address this for 4.2 or even to add hooks which would allow for a price lookup. Please let me know what you guys feel is the primary challenge to getting this added.

Currently my only thought on how to address this issue, and its a hack at best, is to use a custom price rounder that instead of rounding the price, it would fetch the price, given the product and currency from some lookup table and return this value as if it was "rounded". If anyone has any other ideas please let me know - we may be on our own here.

**Comment by jgarniss** [comment]
Are you going to do this change or not? Do I need to go to a new product? There has been no decision announced here one way or another. This is critical to some people.

**Comment by toddca** [comment]
@jgarniss it’s been over 3 years and the core team has not even taken the time to comment or reply to this thread. But they had plenty of time to totally ruin the admin UI in 4.2. The silence speaks volumes.

**Comment by brakim** [comment]
@AndreiMaz stop teasing with remove of on hold label.

**Comment by johansakbas** [comment]
How sure is this to be done in 4.7? Looks very promising if this is added. As mentioned before this is a great deal breaker for many B2B companies to use the platform.

**Comment by orkunsenturk** [comment]
Hope it will be done in 4.7

**Comment by tonywiredin** [comment]
If you want to be a serious e-commerce framework, you need to stop bouncing this to the next version and actually do it

**Comment by brandon71** [comment]
Is this a feature that will ever be supported? A bump to the next revision since 2016 has given us hope but that is fading. It would be great to know if there is any real intention to offer this feature. If not please let us know.

We have clients in multiple countries and every product has their own price for each country. Product prices are based on many factors and we need a way to support client requirements for pricing products.

**Comment by devranerogul** [comment]
Think its going to be added to next version ?

**Comment by xmenxwk** [comment]
> Think its going to be added to next version ?

Nope, its been 7 years, apparently other useless features are more important than this.

**Comment by danFbach** [comment]
I think we all can see the benefits of such a feature set. However, assuming you're all developers, you have to consider that implementation would require a massive rewrite and as developers, you know that minor version updates generally do not include such massive changes. Changes this big will likely introduce an entire new set of potential bugs, a fair number of breaking changes and changes to how an existing store is currently managed. All things any experienced developer would **avoid** in a minor update. 
Assuming these things to be true, I believe it would be most adventegeous to implement this at a major version update (5.x), rather than a minor (4.x) update, so that third party developers can prepare plugin upgrades, store owners have the time to be made aware of interface/management changes, and the code can be **thoroughly** tested. 
Although this issue has been open 7 years, there hasn't been much more than complaining in this thread. Perhaps those desiring the feature and the core maintainers should collaborate on defining a list of milestones with the goal of finally resolving this request and/or start a 5.x alpha branch to encourage community collaboration on bringing this to life.

**Comment by streamzyuk** [comment]
I have been able to extend the code to support multi currency transactions.
Products can have prices in different currencies and and can be set per country.
Products/Categories can be made available to specific countries or be restricted.
Its a little bit hack but the design of the system does not allow the PrimaryStoreCurrency to be set per store or per country.
So I did this by dynamically changing according to customer's country.

I am planning to create a plugin for these features but it will take some time.

**Comment by devranerogul** [comment]
> I have been able to extend the code to support multi currency transactions.
> 
> Products can have prices in different currencies and and can be set per country.
> 
> Products/Categories can be made available to specific countries or be restricted.
> 
> Its a little bit hack but the design of the system does not allow the PrimaryStoreCurrency to be set per store or per country.
> 
> So I did this by dynamically changing according to customer's country.
> 
> 
> 
> I am planning to create a plugin for these features but it will take some time.

Great, looking forward to try plugin.
